cmd/otelcol/main.go

```diff
@@ -65,8 +62,11 @@ func main() {
 	// TODO: Use same format as the collector
 	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
 
-	if !hasFlag("-h") && !hasFlag("--help") && !hasFlag("-v") && !hasFlag("--version") {
-		checkRuntimeParams()
+	// Core flag parser will handle errors, we don't have to handle them here.
+	inputFlags, _ := parseFlags(os.Args[1:])
```

Comment:
Why ignoring error? Shouldn't the program exit and show usage on invalid arguments?

Reply:
Good point. I was originally following existing functionality, but it's much clearer to just directly raise errors.

This is now the message on failure, any feedback is welcome!

```
crobert@ ~/dev/splunk-otel-collector $ ./bin/otelcol --invalid_flag
2022/05/20 15:06:32 main.go:68: Error: flag provided but not defined: -invalid_flag
Use "--help" to show valid usage
```